File: src/main/scala/org/xbery/artbeams/evernote/controller/EvernoteImportController.scala
package org.xbery.artbeams.evernote.controller

import javax.inject.Inject
import javax.servlet.http.HttpServletRequest
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.{GetMapping, RequestMapping}
import org.springframework.web.servlet.ModelAndView
import org.xbery.artbeams.articles.domain.Article
import org.xbery.artbeams.common.controller.{BaseController, ControllerComponents}
import org.xbery.artbeams.evernote.service.EvernoteImporter

/**
  * Evernote import routes.
  * @author Radek Beran
  */
@Controller
@RequestMapping(Array("/admin/evernote"))
class EvernoteImportController @Inject() (evernoteImporter: EvernoteImporter, common: ControllerComponents) extends BaseController(common) {

  @GetMapping(Array("/import"))
  def evernoteImport(request: HttpServletRequest): Any = {
    val articlesOrError = evernoteImporter.importArticles()
    val model = articlesOrError match {
      case Left(ex) =>
        createModel(request, "updatedArticles" -> Seq.empty[Article], "errorMessage" -> ex.toString)
      case Right(articles) =>
        createModel(request, "updatedArticles" -> articles)
    }
    new ModelAndView("/evernote/import", model)
  }
}

File: src/main/kotlin/org/xbery/artbeams/evernote/controller/EvernoteImportController.kt
package org.xbery.artbeams.evernote.controller

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.servlet.ModelAndView
import org.xbery.artbeams.articles.domain.Article
import org.xbery.artbeams.common.controller.BaseController
import org.xbery.artbeams.common.controller.ControllerComponents
import org.xbery.artbeams.evernote.service.EvernoteImporter
import javax.servlet.http.HttpServletRequest

/**
 * Evernote import routes.
 * @author Radek Beran
 */
@Controller
@RequestMapping("/admin/evernote")
open class EvernoteImportController(
    private val evernoteImporter: EvernoteImporter,
    private val common: ControllerComponents
) : BaseController(common) {

    @GetMapping("/import")
    fun evernoteImport(request: HttpServletRequest): Any {
        val model = try {
            val articles = evernoteImporter.importArticles()
            createModel(request, Pair("updatedArticles", articles))
        } catch (ex: Exception) {
            createModel(request, Pair("updatedArticles", listOf<Article>()), Pair("errorMessage", ex.toString()))
        }
        return ModelAndView("/evernote/import", model)
    }
}
